superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts

```diff
@@ -80,9 +82,8 @@ describe('Dashboard filter', () => {
     cy.get('.Select__menu').first().contains('South Asia').click();
 
     // should still have all filter indicators
-    // and since the select is closed, all filter indicators should be visible
     cy.get('svg[data-test="filter"]:visible').should(nodes => {
-      expect(nodes).to.have.length(10);
+      expect(nodes.length).to.least(9);
```

Comment:
there are 10 charts in the dashboard, in which there is 1 filter_box and 9 other charts. How can we expect > 9 filter indicators? (filter_box should never show indictor for itself)

Reply:
<img src="https://user-images.githubusercontent.com/335541/102842559-19c22a80-43bc-11eb-981a-cc283920d88d.png" width="400">

Oh, so it looks the filter indicator expander also has `data-test="filter"`. The test updated it to `10` because `DASHBOARD_NATIVE_FILTERS` is flag is turned on in test config. But obviously this test would fail when the feature flag is disabled.

I think this PR still helps by making the test more lenient. @suddjian wdyt?

Reply:
@suddjian we could also have different test cases for native filter vs non-native filter. And maybe rename the `data-test` for each test element to make them less ambiguous.

Reply:
I agree. I'll try to look into that tomorrow. Do you know if it's possible to turn off the feature flag for a specific test?

Reply:
I'm not sure how to do it or whether it would work, but _maybe_ you could add some common utility to read feature flags from `data-bootrap` of the app container and write different tests based on feature flags.

Reply:
The purpose for this test is _Each chart that applicable to the filter should show filter indicator_, so instead of test the total number of filter indicators, we should check each chart has a filter component. 

Similarly, if there is a test case to see if filter component correctly apply to its scope, then _Only charts in the filter scope should show filter indicator_